app/models/interactive_page.rb

```diff
@@ -301,12 +296,28 @@ def page_number
   def duplicate(helper=nil)
     helper = LaraDuplicationHelper.new if helper.nil?
     new_page = InteractivePage.new(to_hash)
-    new_sections = sections.map { |s| s.duplicate(helper) }
 
     InteractivePage.transaction do
+      new_sections = sections.map { |s| s.duplicate(helper) }
```

Comment:
The fact that is first adds the copied sections to the original page will cause problems in the future. 
Can you pass the new page to the duplicate? So the new sections are created in that new page right from the beginning?
I wonder how this same type of problem is handled when duplicating the page items in a section.

Reply:
@scytacki Do the latest changes look OK? They seem to work correctly in terms of adding the new sections directly to the new page and preventing the need to reset the original sections' positions.

As I understand it, the [`duplicate` method for page items](https://github.com/concord-consortium/lara/blob/new-sections/app/models/page_item.rb#L22-L47) already adds new page items directly to the appropriate section, instead of copying them in place and then moving them if needed.